geoips/commandline/geoips_tree.py

```diff
@@ -0,0 +1,187 @@
+"""GeoIPS CLI "tree" command.
+
+Single 'geoips tree' command which will display GeoIPS CLI commands up to a --max_depth
+value.
+"""
+
+from colorama import Fore, Style
+
+from geoips.commandline.geoips_command import GeoipsExecutableCommand
+
+
+class GeoipsTree(GeoipsExecutableCommand):
+    """Tree Command Class for displaying GeoIPS CLI commands in a tree-like fashion.
+
+    Can be configured to print in a colored fashion, up to a maximum depth, and whether
+    or not we want the full command string or just the command name at it's depth.
+    """
+
+    name = "tree"
+    command_classes = []
+
+    @property
+    def top_level_parser(self):
+        """The parser associated with the top level 'geoips' command."""
+        if not hasattr(self, "_top_level_parser"):
+            # Get 'geoips' parser
+            self._top_level_parser = self.parent.parser
+        return self._top_level_parser
+
+    @property
+    def level_to_color(self):
+        """A color-mapping dictionary based on the depth of the command tree."""
+        if not hasattr(self, "_level_to_color"):
+            self._level_to_color = {
+                0: Fore.RED,
+                1: Fore.YELLOW,
+                2: Fore.BLUE,
+                3: Fore.MAGENTA,
+            }
+        return self._level_to_color
+
+    @property
+    def cmd_line_url(self):
+        """The url to the GeoIPS documentation for the GeoIPS CLI."""
+        if not hasattr(self, "_cmd_line_url"):
+            self._cmd_line_url = r"https://nrlmmd-geoips.github.io/geoips/userguide/command_line.html"  # NOQA
+        return self._cmd_line_url
+
+    def link(self, uri, label=None):
+        """Hyperlink the provided uri alongside 'label' if applicable.
+
+        If label is not provided, just return a hyperlink with 'url' as it's text.
+
+        Parameters
+        ----------
+        uri: str
+            - The uniform resource identifier which maps to a certain webpage.
+        label: str (default = None)
+            - Optional text to hyperlink to.
+
+        Returns
+        -------
+        hyperlink: str
+            - Text that has been hyperlinked to 'uri'
+        """
+        if label is None:
+            label = uri
+
+        parameters = ""
+        # OSC 8 ; params ; URI ST <name> OSC 8 ;; ST
+        escape_mask = "\033]8;{};{}\033\\{}\033]8;;\033\\"
+        hyperlink = escape_mask.format(parameters, uri, label)
+
+        return hyperlink
+
+    def add_arguments(self):
+        """Add arguments to the tree-subparser for the Tree Command."""
+        self.parser.add_argument(
+            "--max_depth",
+            default=2,
+            type=int,
+            choices=[0, 1, 2],
```

Comment:
I don't think that we need choices here. Just accept any integer. If the integer is greater than the number of available levels, just print what is available. So, `--max-depth 4` should result in the same output as `--max-depth 2` given the current CLI.

Reply:
Removed choices and added a check in the call function to confirm that the max depth is in the valid range.